modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/LineMBLayer.java

```diff
@@ -281,19 +268,8 @@ public Point lineTranslate() {
      *
      * @return The geometry's offset, as a Displacement.
      */
-    public Displacement toDisplacement() {
-        Object defn = paint.get("line-translate");
-        if (defn == null) {
-            return null;
-        } else if (defn instanceof JSONArray) {
-            JSONArray array = (JSONArray) defn;
-            return sf.displacement(parse.number(array, 0, 0), parse.number(array, 1, 0));
-        } else if (defn instanceof JSONObject) {
-            throw new MBFormatException("\"line-translate\": Functions not supported yet.");
-        } else {
-            throw new MBFormatException("\"line-translate\": Expected array or function, but was "
-                    + defn.getClass().getSimpleName());
-        }
+    public Displacement lineTranslateDisplacement() {
```

Comment:
That would be fine. The difference is that most layers only have one displacement -- Symbol layer has several.